webapp/packages/plugin-connections/src/ConnectionForm/Options/useOptions.ts

```diff
@@ -32,13 +31,10 @@ export function useOptions(state: IConnectionFormState) {
     updateNameTemplate(driver: DBDriver | undefined) {
       runInAction(() => {
         const {
-          prevName,
           state: { config, info, mode },
         } = refObject;
 
-        const isAutoFill = config.name === prevName || prevName === null;
-
-        if (mode === 'edit' || !isAutoFill) {
+        if (mode === 'edit') {
```

Comment:
it seems like it will reset the name to default every time driver is changed even custom name is set